CMakeLists.txt

```diff
@@ -219,7 +219,9 @@ option(USE_TENSORRT "Using Nvidia TensorRT library" OFF)
 cmake_dependent_option(
     USE_VALGRIND "Use Valgrind. Only available on Linux." ON
     "LINUX" OFF)
-option(USE_VULKAN "Use Vulkan GPU backend" OFF)
+cmake_dependent_option(
+    USE_VULKAN "Use Vulkan GPU backend" ON
+    "ANDROID" OFF)
```

Comment:
Is USE_VULKAN dependent on ANDROID being OFF or ON?  Will this be OK for desktop builds?

Reply:
https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html
Here the argument positions are non trivial: the default value is the latest, ON if ANDROID

Reply:
Got your reply in an email, so is my understanding correct that this statement means: give the users the option to set USE_VULKAN to ON or OFF if ANDROID is OFF, right?

Reply:
Yes, I deleted my reply to verify if overrides with -DUSE_VULKAN work correct.

And in that case it does not. It is ON if android and OFF else.

I will try to find the way how to make override work

Reply:
@AshkanAliabadi , updated to construction
```
if(NOT DEFINED USE_VULKAN)
  cmake_dependent_option(
      USE_VULKAN "Use Vulkan GPU backend" ON
      "ANDROID" OFF)
endif()
```
which works fine with -DUSE_VULKAN=